rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/transform/TbChangeOriginatorNodeTest.java

```diff
@@ -20,134 +20,343 @@
+    }
+
+    @Test
+    public void givenEntitySourceAndEntityTypeIsNull_whenInit_thenThrowsException() {
+        config.setOriginatorSource(ENTITY_SOURCE);
+        config.setEntityType(null);
+
+        assertThatThrownBy(() -> node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config))))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("Entity type should be specified if 'Entity by name pattern' source is selected.");
+    }
+
+    @ParameterizedTest
+    @NullAndEmptySource
+    public void givenEntitySourceAndEntityNamePatternIsEmpty_whenInit_thenThrowsException(String entityName) {
+        config.setOriginatorSource(ENTITY_SOURCE);
+        config.setEntityType(EntityType.DEVICE.name());
+        config.setEntityNamePattern(entityName);
 
-        assertEquals(customerId, originatorCaptor.getValue());
+        assertThatThrownBy(() -> node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config))))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("Name pattern should be specified if 'Entity by name pattern' source is selected.");
     }
 
     @Test
-    public void exceptionThrownIfCannotFindNewOriginator() {
-        AssetId assetId = new AssetId(Uuids.timeBased());
-        CustomerId customerId = new CustomerId(Uuids.timeBased());
-        Asset asset = new Asset();
-        asset.setCustomerId(customerId);
+    public void givenEntitySourceAndUnexpectedEntityType_whenInit_thenThrowsException() {
+        config.setOriginatorSource(ENTITY_SOURCE);
+        config.setEntityType(EntityType.TENANT.name());
+        config.setEntityNamePattern("tenant-A");
+
+        assertThatThrownBy(() -> node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config))))
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessage("Unexpected entity type TENANT");
+    }
 
-        RuleChainId ruleChainId = new RuleChainId(Uuids.timeBased());
-        RuleNodeId ruleNodeId = new RuleNodeId(Uuids.timeBased());
+    @Test
+    public void givenOriginatorSourceIsCustomer_whenOnMsg_thenTellSuccess() throws TbNodeException {
+        Device device = new Device(DEVICE_ID);
+        device.setCustomerId(CUSTOMER_ID);
+
+        TbMsg msg = TbMsg.newMsg(TbMsgType.POST_TELEMETRY_REQUEST, DEVICE_ID, TbMsgMetaData.EMPTY, TbMsg.EMPTY_JSON_OBJECT);
+        TbMsg expectedMsg = TbMsg.transformMsgOriginator(msg, CUSTOMER_ID);
+
+        given(ctxMock.getDbCallbackExecutor()).willReturn(dbExecutor);
+        given(ctxMock.getDeviceService()).willReturn(deviceServiceMock);
+        given(ctxMock.getTenantId()).willReturn(TENANT_ID);
+        given(deviceServiceMock.findDeviceById(any(TenantId.class), any(DeviceId.class))).willReturn(device);
+        given(ctxMock.transformMsgOriginator(any(TbMsg.class), any(EntityId.class))).willReturn(expectedMsg);
```

Comment:
please add a verification to check what will be passed to the transformMsgOriginator method. Add this in other places as well.